screens/Contacts/ContactGraph/ContactGraph.tsx

```diff
@@ -1,12 +1,95 @@
-import { View, Text } from "react-native"
+import { View, TextInput, TouchableWithoutFeedback, Keyboard } from "react-native"
 import { styles } from "./styles"
+import Graph, { Node } from "../../../components/Graph/Graph"
 
-const ContactGraph = () => {
+import ForceDirectedGraph from "../../../components/Graph/ForceDirectedGraph/ForceDirectedGraph"
+import { useState } from "react"
+
+import NodeModal from "../../../components/Graph/NodeModal/NodeModal"
+
+const graph = new Graph(
+  ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"],
+  ["0", "2", "3", "4", "5", "6", "7", "8", "9", "10"],
+  ["1", "1", "1", "1", "1", "1", "2", "2", "3", "3"],
+  [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5]
+)
+const constrainedNodeId = "1"
+
+interface ContactGraphProps{
+  onContactPress: (uid : string) => void
+}
+
+const ContactGraph = ({onContactPress} : ContactGraphProps) => {
+
+  const [searchText, setSearchText] = useState("")
+
+  const [clickedNode, setClickedNode] = useState<Node >(graph.getNodes()[0])
+
+  const [modalVisible, setModalVisible] = useState(false)
+
+  const onModalPress = (uid: string) => {
+    const node = graph.getNodeById(uid)
+    if (node) {
+      setClickedNode(node)
+      setModalVisible(true)
+    }
+  }
+
+  const onPressOut = () => {
+    setModalVisible(false)
+  }
   return (
+    
+    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
     <View style={styles.container}>
-      <Text>Graph view</Text>
+      <TextInput
+        style={styles.searchBar}
+        placeholder="Search..."
+        value={searchText}
+        onChangeText={(text) => {
+          setSearchText(text)
+          handleSearch(text, graph)
+         }
+        }
+        onSubmitEditing={() => handleQuery(onContactPress)}
+      />
+    
+    <NodeModal node={clickedNode} visible={modalVisible} onPressOut={onPressOut} onContactPress={onContactPress} />
+    <View style={styles.graphContainer}>
+
+    <ForceDirectedGraph graph={graph} constrainedNodeId={constrainedNodeId} onModalPress={onModalPress} />
     </View>
+    </View>
+    </TouchableWithoutFeedback>
   )
 }
 
 export default ContactGraph
+
+function handleSearch(text: string, graph: Graph): void {
+  if (text === "") {
+    for (const node of graph.getNodes()) {
+      node.selected = false
+    }
+  }
+  else {
+    for (const node of graph.getNodes()) {
+      if (node.id.includes(text) || text.includes(node.id)) {
```

Comment:
Here you had a TODO saying to use regex for text search. I think this should be implemented sometimes. Maybe you can add an issue for that.

Reply:
Yes I had one but eslintrc did not let me keep it